File: objects/src/main/kotlin/objectsData/Castle.kt
package objectsData

import enums.ObjectTypeEnum
import kotlinx.serialization.EncodeDefault
import kotlinx.serialization.ExperimentalSerializationApi

@OptIn(ExperimentalSerializationApi::class)
@kotlinx.serialization.Serializable
data class Castle(
    override val id: Int,
    override var loc: Loc,
    val soldierType: ObjectTypeEnum,
    @EncodeDefault(EncodeDefault.Mode.ALWAYS) override val type: ObjectTypeEnum = ObjectTypeEnum.CASTLE
) : GameObject()

File: runner/src/main/resources/apis/ts/src/classes/objects/Castle.ts
import GameObject from "./GameObject";
import Loc from "../Loc";

export default  class Castle extends GameObject {
    soldierType: string

    constructor(id: number, loc: Loc, soldierType: string) {
        super("CASTLE", id, loc);
        this.soldierType = soldierType;
    }
}